src/command_line/substructure_search.rs

```diff
@@ -32,61 +33,86 @@ pub fn command() -> Command {
 
 pub fn action(matches: &ArgMatches) -> eyre::Result<()> {
     let index_path = matches.get_one::<String>("index").unwrap();
-    let smiles = matches.get_one::<String>("smiles").unwrap();
-    let limit = matches.get_one::<String>("limit");
+    let smile = matches.get_one::<String>("smiles").unwrap();
+    let result_limit = matches.get_one::<String>("limit");
 
-    // Validate structure
-    let problems = validate_structure(smiles)?;
-    if !problems.is_empty() {
-        panic!("Need to do something here to either correct the smiles query or return an error message");
-    }
-
-    // Need to process cpd ahead of time as we may need to iterate through tautomers later
-    let (query_canon_taut, fingerprint, descriptors) = process_cpd(smiles).unwrap();
+    let (query_canon_taut, fingerprint, descriptors) = prepare_query_structure(smile)?;
 
     let index = open_index(index_path)?;
     let reader = index.reader()?;
     let searcher = reader.searcher();
 
-    let limit = if let Some(limit) = limit {
-        limit.parse::<usize>()?
+    let result_limit = if let Some(result_limit) = result_limit {
+        result_limit.parse::<usize>()?
     } else {
         usize::try_from(1000)?
     };
 
-    let tantivy_result_limit = limit * 10;
+    let tautomer_limit = &result_limit * 10;
```

Comment:
Try running `cargo clippy` and see if it complains about this line. `result_limit` is a usize which is "trivially" copyable in rust speak so you wouldn't bother with a reference. The reference will be 64bit and the usize is 64bit too so it's just extra indirection.

Reply:
Forgot all about clippy... Should be all set now!